dali/operators/segmentation/random_object_bbox.cc

```diff
@@ -424,22 +503,24 @@ void RandomObjectBBox::RunImpl(HostWorkspace &ws) {
   TensorShape<> default_anchor;
   default_anchor.resize(ndim);
 
-  contexts_.resize(tp.size()+1);
+  default_context_.thread_pool = &tp;
+  huge_context_.thread_pool = &tp;
 
   std::uniform_real_distribution<> foreground(0, 1);
   for (int i = 0; i < N; i++) {
     bool fg = foreground(rngs_[i]) < foreground_prob_[i].data[0];
     if (!fg) {
       StoreBox(out1, out2, format_, i, default_anchor, input.tensor_shape(i));
       if (HasClassLabelOutput()) {
-        SampleContext &ctx = contexts_[0];
+        auto &ctx = default_context_;
         GetBgFgAndWeights(ctx.classes, ctx.weights, ctx.background, i);
         class_label_out.data[i][0] = ctx.background;
       }
     } else {
-      tp.AddWork([&, i](int thread_idx) {
-        SampleContext &ctx = contexts_[thread_idx+1];
-        ctx.Init(i, &input[i]);
+      auto blob_label = (input[i].size() > 0x80000000L) ? DALI_INT64 : DALI_INT32;
```

Comment:
It'd be more readable if you use numeric_limits or similar, instead of the literal